device/pcie/pci_device.hpp

```diff
@@ -34,84 +34,144 @@ struct PciDeviceInfo
 };
 
 class PCIDevice {
+    const std::string device_path;  // Path to character device: /dev/tenstorrent/N
+    const int device_num;           // N in /dev/tenstorrent/N
+    const int logical_id;           // Unique identifier for each device in entire network topology
+    const int fd;                   // Character device file descriptor
+    const PciDeviceInfo info;       // PCI device info
+    const int numa_node;            // -1 if non-NUMA
+    const int revision;             // PCI revision value from sysfs
+    const tt::ARCH arch;            // e.g. Grayskull, Wormhole, Blackhole
+    std::unique_ptr<tt::umd::architecture_implementation> architecture_implementation;
+
 public:
     /**
-     * Return a list of integers corresponding to character devices in /dev/tenstorrent/
+     * @return a list of integers corresponding to character devices in /dev/tenstorrent/
      */
     static std::vector<int> enumerate_devices();
 
-    PCIDevice(int device_id, int logical_device_id);
+    /**
+     * PCI device constructor.
```

Comment:
Great docs. You can maybe add these functions to doxygen docs list, I think these are undocumented at the moment

Reply:
Fixed - although this was a little hard to find so I updated `docs/README.md`